app/server-utils/url-to-iframe.ts

```diff
@@ -1,29 +1,34 @@
 /**
- * Replaces all of the URLs in text with iframes of the URLs
+ * Replaces all whitelisted URLs in the text with iframes of the URLs
  * @param text text in which URLs should be replaced
  */
 export const urlToIframe = (text: string): string => {
   const whitelistedHosts: Record<string, HostConfig> = {
     'aisafety.world': {sandboxValue: 'allow-scripts allow-same-origin'},
   }
 
-  // Regex is from: https://stackoverflow.com/a/26764609
-  const anchorTagRegex = new RegExp(/<a[\s]+([^>]+)>((?:.(?!<\/a>))*.)<\/a>/gi)
-  const matches = text.matchAll(anchorTagRegex)
+  // Regex to extract known links:
+  // \[               <-- start of markdown link, i.e. [name](url)
+  //   (              <-- capture the url
+  //     https?:\/\/  <-- check for both http and https
+  //     [\.\w]+?     <-- any string of alphanumerical characters and full stops
+  //     \/?[^\]]*    <-- optional query params
+  //   )
+  // \]
+  // \(
+  //     \1           <-- lookback to match whatever the first group caught
+  // \)
+  const linkRegex = new RegExp(/\[(https?:\/\/[.\w]+?\/?[^]]*?)\]\(\1\)/gi)
```

Comment:
this wasn't working for me, so I refactored it to work on markdown

Reply:
if using regex literal `/.../gi`, the `new RegExp(string)` wrapper is not needed - please choose one of the syntaxes, not both